src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin/Microsoft.Subscriptions/ProviderRegistrations/AddResourceProviderRegistration.cs

```diff
@@ -136,8 +138,19 @@ protected override object ExecuteCore()
                                 RoutingResourceManagerType = this.ResourceManagerType,
                                 Enabled = true,
                                 ProviderLocation = this.ProviderLocation,
-                                ExtensionName = this.ExtensionName,
-                                ExtensionUri = this.ExtensionUri.AbsoluteUri,
+                                // Todo: Remove this HACK, hardcoded versions to have backward compatibility 
+                                ExtensionCollection = new ExtensionCollectionDefinition()
```

Comment:
@markcowl 
This is for registering the resource provider with azurestack. In azure this is a manual admin process and they have .exe to do this. In azurestack to register a resource provider like compute RP at the deployment, this cmdlet is used. This will also be used by third party RP developers

Here we have gone through a breaking change in the back end to add additional properties for resource provider ui extensions.  we don't want to break the cmdlet and third parties for a release and will remove the backward compat in future (after GA)